app/models/manageiq/providers/ibm_cloud/inventory/collector/power_virtual_servers.rb

```diff
@@ -102,6 +103,11 @@ def sap_profiles
     @sap_profiles ||= sap_api.pcloud_sap_getall(cloud_instance_id).profiles || []
   end
 
+  def server_groups
+    byebug
+    placement_groups_api.pcloud_placementgroups_getall(cloud_instance_id) || []
```

Comment:
Recommend caching this so it can be used again without re-issuing the API call:
```suggestion
    @server_groups ||= placement_groups_api.pcloud_placementgroups_getall(cloud_instance_id) || []
```

Reply:
Then you can do:
```ruby
def server_groups_by_id
  server_groups.index_by(&:id)
end
```
if you want, to replace https://github.com/ManageIQ/manageiq-providers-ibm_cloud/pull/363/files#diff-d244b92097e187e59c19aceb5f90f45f0de129bb130fa3d8d0cd1682a2e686e4R56-R59

You might not need it though, don't see it used anywhere